client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,217 @@
+<script lang="ts">
+  import type { PrivatePlayer, Lobby } from "../../../functions/src/firestore-types/lobby";
+  import { onMount } from "svelte";
+  import { lobbyReturn } from "$lib/firebase/firebase-functions";
+  import { authStore as user } from "$stores/auth";
+
+  export let lobbyCode: string;
+  export let lobby: Lobby;
+  export let privatePlayer: PrivatePlayer;
+
+  let end: "Cat Win" | "Cat Lose" | "Catfish Lose" | "Catfish Win" | null = null;
+  let catfishList: string = "";
+  let catfishes: string[];
+  let isHost: boolean;
+  onMount(() => {
+    // check if the user is the host
+    if ($user !== null) {
+      if (lobby.uids.indexOf($user.uid) == 0) {
+        isHost = true;
+      } else {
+        isHost = false;
+      }
+    }
+    // format a string list to have commas and "and" when needed
+    const formatter = new Intl.ListFormat("en", { style: "long", type: "conjunction" });
+    // get the list of catfishes from lobby.ts and put them into a variable
+    catfishes = lobby.players
+      .filter((p) => {
+        return p.role == "CATFISH";
+      })
+      .map((c) => c.displayName);
+    catfishList = formatter.format(catfishes);
+
+    // grab the current user's role
+    const myRole = privatePlayer.role;
+    // update end variable depending on the winner type and the current player's role
+    if (myRole == "CAT") {
+      if (lobby.winner == "CAT") {
+        end = "Cat Win";
+      } else if (lobby.winner == "CATFISH") {
+        end = "Cat Lose";
+      }
+    } else {
+      if (lobby.winner == "CAT") {
+        end = "Catfish Lose";
+      } else if (lobby.winner == "CATFISH") {
+        end = "Catfish Win";
+      }
+    }
```

Comment:
should check if `lobby.winner` is `undefined` before so you can just have if/else instead of elseif